src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3287,7 +3287,7 @@ void MacroAssembler::lookup_secondary_supers_table(Register r_sub_klass,
   z_bru(L_done); // pass whatever result we got from a slow path
 
   bind(L_failure);
-  // TODO: use load immediate on condition and z_bru above will not be required
+
   z_lghi(r_result, 1);
```

Comment:
This lghi deserves a conversion to locghi. Furthermore, without checking for feature availability, the code will require z13 or newer. Is that intended?

Reply:
yeah, actually that was my intention but then I find out that codepath which jumps to `L_failure` not comes with setting cc to 2. 

```
  testbit(r_array_index, 63);
  z_bfalse(L_failure);
```

```
  testbit(r_bitmap, (bit + 1) & Klass::SECONDARY_SUPERS_TABLE_MASK);
  z_bfalse(L_failure);
```

So other solution which I could think of is, to assume that search will fail and load `1` at the start 
```
  // Initialize r_result with 0 (indicating success). If searching fails, r_result will be loaded
  // with 1 (failure) at the end of this method.
  clear_reg(r_result, true /* whole_reg */, false /* set_cc */); // r_result = 0
```

But then I remembered this: https://github.com/openjdk/jdk/pull/19544/files#r1660693537

Reply:
>So other solution which I could think of is, to assume that search will fail and load 1 at the start 

with this we can now look for `CC = EQ`. And use `z_locghi(result, 0, Assembler::bcondEqual);`

Also now logic for secondary supers table for c1 & interpreter assumes that search will fail and loads 1 at the start. So to be in sync, better we keep it this way :)